packages/types/src/codec/Compact.ts

```diff
@@ -20,13 +20,13 @@ import UInt from './UInt';
  * used by other types to add length-prefixed encoding, or in the case of wrapped types, taking
  * a number and making the compact representation thereof
  */
-export default class Compact extends Base<UInt | Moment> implements Codec {
-  public constructor (Type: Constructor<UInt | Moment>, value: Compact | AnyNumber = 0) {
-    super(Compact.decodeCompact(Type, value));
+export default class Compact<T> extends Base<UInt | Moment> implements Codec {
```

Comment:
Need some help here - this does not look 100% -

- Base needs to use `T` as well
- All `UInt | Moment` to be `T`

... however, cannot quite get all the checkers happy, so the current attempt is better, but far from technically correct. (Please pull down, play here and give suggestions - or just push a proper fix)

Reply:
Thanks to @amaurymartiny a fix is in :)